controller/src/beerocks/bml/bml.cpp

```diff
@@ -709,3 +709,48 @@ int bml_start_dcs_single_scan(BML_CTX ctx, const char *radio_mac, int dwell_time
     return pBML->start_dcs_single_scan(tlvf::mac_from_string(std::string(radio_mac)), dwell_time,
                                        channel_pool, channel_pool_size);
 }
+
+int bml_client_get_client_list(BML_CTX ctx, char *client_list, unsigned int *client_list_size)
+{
+    // Validate input parameters
+    if (!ctx || !client_list || !client_list_size) {
+        return (-BML_RET_INVALID_ARGS);
+    }
+
+    if (*client_list_size == 0) {
+        return (-BML_RET_INVALID_ARGS);
+    }
+
+    auto pBML = static_cast<bml_internal *>(ctx);
+    std::string temp_client_list;
+    int ret = pBML->client_get_client_list(temp_client_list, client_list_size);
+    if (ret == BML_RET_OK) {
+        beerocks::string_utils::copy_string(client_list, temp_client_list.c_str(),
+                                            *client_list_size);
+    }
+
+    return ret;
+}
+
+int bml_client_set_client(BML_CTX ctx, const char *sta_mac,
+                          const struct BML_CLIENT_CONFIG *client_config)
+{
+    // Validate input parameters
+    if (!ctx) {
+        return (-BML_RET_INVALID_ARGS);
+    }
+
+    auto pBML = static_cast<bml_internal *>(ctx);
+    return pBML->client_set_client(tlvf::mac_from_string(std::string(sta_mac)), *client_config);
+}
+
+int bml_client_get_client(BML_CTX ctx, const char *sta_mac, struct BML_CLIENT *client)
+{
+    // Validate input parameters
+    if (!ctx && !client) {
+        return (-BML_RET_INVALID_ARGS);
+    }
+
+    auto pBML = static_cast<bml_internal *>(ctx);
+    return pBML->client_get_client(tlvf::mac_from_string(std::string(sta_mac)), client);
```

Comment:
@morantr is there an added value for the explicit std::string here?
or would this suggestion be valid? (implicit conversion to uint8_t and call to `std::string mac_to_string(const uint8_t *mac_address);`)
```suggestion
    return pBML->client_get_client(tlvf::mac_from_string(sta_mac), client);
```

this comment refers to all occurrences

Reply:
The main issue is that the BML receives an ascii char array, meaning an 18 byte array containing the string representation of the mac. which will mean that `std::string mac_to_string(const uint8_t *mac_address)` wouldn't work here.
I tried using without the explicit std::string conversion, but got errors when I did.
This seems like the simplest solution.

A better one would be to expand `tlvf::mac_from_string` and add another overload function that receives an ascii represented char array like so: `sMacAddr tlvf:mac_from_string(const char[18] buf)`
And in this new overload the would simply trigger the original `sMacAddr rlvf:mac_from_string(const std::string &)` after converting the buffer to an std:string

Reply:
if that is the case, your approach is ok for now - might need to add the function overloading you suggested but not as part of this PR